src/looprestoration.rs

```diff
@@ -1719,3 +1741,130 @@ unsafe extern "C" fn sgr_filter_mix_neon<BD: BitDepth>(
         bd,
     );
 }
+
+#[cfg(all(feature = "asm", any(target_arch = "x86", target_arch = "x86_64")))]
+#[inline(always)]
+fn loop_restoration_dsp_init_x86<BD: BitDepth>(
+    c: &mut Dav1dLoopRestorationDSPContext,
+    bpc: libc::c_int,
+) {
+    use crate::src::x86::cpu::*;
+
+    let flags = dav1d_get_cpu_flags();
+
+    if flags & DAV1D_X86_CPU_FLAG_SSE2 == 0 {
+        return;
+    }
+
+    #[cfg(feature = "bitdepth_8")]
+    if BD::BPC == BPC::BPC8 {
+        c.wiener[0] = dav1d_wiener_filter7_8bpc_sse2;
+        c.wiener[1] = dav1d_wiener_filter5_8bpc_sse2;
+    }
+
+    if flags & DAV1D_X86_CPU_FLAG_SSSE3 == 0 {
+        return;
+    }
+
+    c.wiener[0] = dav1d_wiener_filter7_ssse3::<BD>;
+    c.wiener[1] = dav1d_wiener_filter5_ssse3::<BD>;
+
+    if BD::BPC == BPC::BPC8 || bpc == 10 {
+        c.sgr[0] = dav1d_sgr_filter_5x5_ssse3::<BD>;
+        c.sgr[1] = dav1d_sgr_filter_3x3_ssse3::<BD>;
+        c.sgr[2] = dav1d_sgr_filter_mix_ssse3::<BD>;
+    }
+
+    #[cfg(target_arch = "x86_64")]
+    {
+        if flags & DAV1D_X86_CPU_FLAG_AVX2 == 0 {
+            return;
+        }
+
+        c.wiener[0] = dav1d_wiener_filter7_avx2::<BD>;
+        c.wiener[1] = dav1d_wiener_filter5_avx2::<BD>;
+
+        if BD::BPC == BPC::BPC8 || bpc == 10 {
+            c.sgr[0] = dav1d_sgr_filter_5x5_avx2::<BD>;
+            c.sgr[1] = dav1d_sgr_filter_3x3_avx2::<BD>;
+            c.sgr[2] = dav1d_sgr_filter_mix_avx2::<BD>;
+        }
+
+        if flags & DAV1D_X86_CPU_FLAG_AVX512ICL == 0 {
+            return;
+        }
+
+        c.wiener[0] = dav1d_wiener_filter7_avx512icl::<BD>;
+        c.wiener[1] = match BD::BPC {
+            // With VNNI we don't need a 5-tap version.
+            BPC::BPC8 => c.wiener[0],
+
+            #[cfg(feature = "bitdepth_16")]
+            BPC::BPC16 => dav1d_wiener_filter5_16bpc_avx512icl,
+
+            #[cfg(not(feature = "bitdepth_16"))]
+            BPC::BPC16 => unreachable!(),
```

Comment:
I think we can do this without an `unreachable!` arm.  For my previous matches on `BPC`, I didn't have a cfg and it still handled everything correctly when only compiling for one bitdepth.